select.c

```diff
@@ -177,12 +178,30 @@ select_dispatch(struct event_base *base, struct timeval *tv)
 	check_selectop(sop);
 
 	if (res == -1) {
-		if (errno != EINTR) {
-			event_warn("select");
-			return (-1);
+		if (errno == EINTR) {
+		        return (0);
 		}
-
-		return (0);
+                /* There seems to be a very subtle race condition between the
+                 * event_del and the select, where the fd is still active on the
+                 * event_readset_in but no libevent structure make reference
```

Comment:
"no libevent structure makes reference to it."

Remove that "so it" and s/make/makes/